Contributor Guide format/structure updates

PeterTurcan opened the pull request:
Round of formatting and structure updates to the Contributor Guide. Removed some duplicate files/topics, updated some asciidocs formatting. Improved consistency.

alandefreitas commented on contributor-guide/modules/ROOT/pages/design-guide/headers.adoc:
```diff
@@ -4,57 +4,40 @@
 
 @anarthal: this is taken verbatim from old content. It may be outdated. Conversion to asciidoc is not complete.
```
We should replace all these `anarthal` comments with small admonitions since the page is already live.

PeterTurcan replied:
Agreed, move all the comments to a separate doc, replaced with short sentence as appropriate:
*Content under construction.*
*Legacy content, currently being updated.*

PeterTurcan commented on contributor-guide/modules/ROOT/nav.adoc:
```diff
@@ -5,12 +5,10 @@
 
-* xref:design-guide/intro.adoc[]
-** xref:design-guide/code-design.adoc[]
+* xref:design-guide/code-design.adoc[]
```
Agreed, done.

PeterTurcan commented on contributor-guide/modules/ROOT/pages/design-guide/code-design.adoc:
```diff
@@ -13,8 +13,9 @@ Please use these guidelines as a checklist for preparing the content a library s
+
+  . More Effective C++
 +
 both by Scott Meyers, published by Addison Wesley.
```
rewritten

PeterTurcan commented on contributor-guide/modules/ROOT/pages/design-guide/code-design.adoc:
```diff
@@ -13,8 +13,9 @@ Please use these guidelines as a checklist for preparing the content a library s
 
 * Follow quality programming practices. For example:
```
agreed, added more sections and structure

PeterTurcan commented:
@alandefreitas Addressed your comments, and restructured the nav file to match the User Guide (Headings are just text and not themselves links to files).

alandefreitas commented:
Great! Please squash :)
